lib/awful/prompt.lua

```diff
@@ -285,10 +289,14 @@ local function prompt_text_with_cursor(args)
         text_start = gstring.xml_escape(text)
         text_end = ""
     else
-        char = gstring.xml_escape(text:sub(args.cursor_pos, args.cursor_pos))
+        local offset = 0
+        if have_multibyte_char_at(text, args.cursor_pos) then
+            offset = 1
```

Comment:
can't utf8 sequences take up to four bytes (or even more)? This seems to only handle one or two bytes.

Reply:
previous code was handling only one-byte

i suppose users of 3-byte languages could also make the changes later on

Reply:
Yeah. A single code point can be up to four bytes long, and certain characters are built from two consecutive code points.

I can confirm that this breaks with 3-byte or 4-byte characters (tested with U+1000 `က` and U+10000 `𐀀`):
```
Error parsing markup: Error on line 1 char 83: Invalid UTF-8 encoded text in name — not valid “� '
stack traceback:
        /home/lucas/projects/awesome/lib/wibox/widget/textbox.lua:193: in function 'wibox.widget.textbox.set_markup'
        /home/lucas/projects/awesome/lib/awful/prompt.lua:566: in upvalue 'update'
        /home/lucas/projects/awesome/lib/awful/prompt.lua:927: in local 'keygrabber_function'
        /home/lucas/projects/awesome/lib/awful/keygrabber.lua:146: in function </home/lucas/projects/awesome/lib/awful/keygrabber.lua:143>
```

The ranges for single code points are

| First code point  | Last code point  | Length  |
|:-----------------:|:----------------:|---------|
|           U+0000  |          U+007F  | 1 Byte  |
|           U+0080  |          U+07FF  | 2 Bytes |
|           U+0800  |          U+FFFF  | 3 Bytes |
|          U+10000  |        U+10FFFF  | 4 Bytes |

I would assume that the set of characters that are built from multiple code points is defined as well.

Reply:
also, as i mentioned in previous comments, this PR implement multibyte support only for some of the keybindings, so it gonna serve as an example, for people who would want to either add multibyte support for more prompt hotkeys or more bytes

btw, the pcall thingy wasnt performing well enough even as a workaround -- update() function is called from several places, and pcall-ing one of them was hovering only some of the prompt hotkeys

Reply:
to make it clear - i'm not going to debug tests `Assertion failed □ is not equal to □`